### DISCUSSION TITLE
each directive creates two caches

### DISCUSSION THREAD

**Discussion by Askancy** [ryangjchandler/blade-cache-directive#12] [Q&A] [answered]
I created in a blade three separate cache directives, more precisely:

`@cache("article_{$article->id}")`

`@cache("article_related_{$article->id}")`

`@cache("widget_game_{$giochi->id}")`

checking the debugbar though, these directives are created twice:

![image](https://github.com/ryangjchandler/blade-cache-directive/assets/789702/ad503dad-4d1e-49c0-90ca-79d1cb3ab9c5)

caches are created by controller with Laravel directive, caches created by this package instead are duplicated... how come?

1 x hit article_sql_cyberpunk-laggiornamento-21-e-lultimate-edition-sono-disponibili-da-oggi.html (0%)
1 x hit setting (0%) 

I can't create a issue ticket, the links lead to a 404 error

**Comment by ryangjchandler** [comment]
Please share the code fully.

**Comment by Askancy** [comment]
Find the cache here @cache("article_{$article->id}") and @cache("article_related_{$article->id}"):
```
@section('content')

@cache("article_{$article->id}")

<section>
  <div class="container mx-auto">
    <div class="bg-white dark:bg-primaryDark-200 py-4">

      <div class="flex flex-wrap md:mb-6">
        <div class="lg:w-3/4 md:w-3/5 w-full">
          <article class="art_ge" id="gallery">

              <h2 class="mt-5">
                {!! $article->titolo !!}
                  @if (\IPS\Member::loggedIn()->isAdmin() OR \IPS\Member::loggedIn()->modPermission())
                    <span><a href="{{ url('admin/articoli/'.$article->id.'/edit')}}" target="_blank" wire:navigate.hover>{{ __('public.edit') }}</a></span>
                  @endif
              </h2>

        @if (config('app.debug') === 'false')
          {!! parselink($testo) !!}
        @else
          {!! $testo !!}
        @endif


        </article>
@endcache

        @if (!empty($article->tag))
          @cache("article_related_{$article->id}")
            <livewire:related-articles :articleId="$article->id" />
          @endcache
        @endif

        @include('front.pages.user.comment.new')
        @include('front.pages.user.comment.reply')

      </div>
      <div class="lg:w-1/4 md:w-2/5 w-full px-4">
        @if($article->gioco_principale > '0')
          @include('front.component.elements.widget-game', ['giochi' => $article->giochi, 'gioco' =>  $article->gioco_p, 'info_giochi' => $article->infogiochi])
        @endif
      </div>
    </div>
  </div>
  </div>
</section>
@endsection
```

@cache("widget_game_{$giochi->id}"), Instead, it is located inside the include "elements.widget-game".

**Reply by ryangjchandler** [reply, marked as answer]
So I think you're actually reading the Debugbar output wrong.. it says that the cache key was "hit" twice, meaning it was read from twice, not created. This is because we first check to see if the cache key exists, and then retrieve the value from the cache. Creation only ever happens once in code, therefore there aren't any problems here.
